torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -502,6 +502,7 @@ def aot_dispatch_autograd(
                     and fw_metadata.input_info[i].mutates_storage_metadata
                 )
                 if bw_out is None and not metadata_mutation_in_graph:
+                    continue  # HACK(yf225): please replace with the right fix
```

Comment:
Need to replace with the right fix, for the `requires_grad=False` bwd graph input duplication issue.

Reply:
hmm can you change this to:
```
is_non_leaf = fw_metadata.input_info[i].requires_grad and not fw_metadata.input_info[i].is_leaf
if bw_out is None and not metadata_mutation_in_graph and is_non_leaf:
    ...
```

can you also split this one out into a separate PR?

Reply:
will use your PR https://github.com/pytorch/pytorch/pull/134193 :D